frontend/src/common/ReviewWithComment.tsx

```diff
@@ -1,45 +1,123 @@
-import { Button, Dialog, DialogContent, DialogContentText, DialogTitle, Stack, TextField } from '@mui/material'
-import { useState } from 'react'
+import {
+  Autocomplete,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Stack,
+  TextField,
+} from '@mui/material'
+import { useEffect, useState } from 'react'
+
+import { useGetModelRoles } from '../../actions/model'
+import { useGetReviewRequestsForModel } from '../../actions/review'
+import { ReviewRequestInterface } from '../../types/interfaces'
+import { ReleaseInterface } from '../../types/types'
+import { getRoleDisplay } from '../../utils/beta/roles'
+import MessageAlert from '../MessageAlert'
+import Loading from './Loading'
 
 type ReviewWithCommentProps = {
   open: boolean
   onClose: () => void
-  onSubmit: (kind: ResponseTypeKeys, reviewComment: string) => void
+  onSubmit: (kind: ResponseTypeKeys, reviewComment: string, reviewRole: string) => void
   title: string
   description?: string
+  release: ReleaseInterface
 }
 
 export const ResponseTypes = {
   Approve: 'approve',
-  RequestChanges: 'request changes',
-  Reject: 'reject',
+  RequestChanges: 'request_changes',
 } as const
 
 export type ResponseTypeKeys = (typeof ResponseTypes)[keyof typeof ResponseTypes]
 
-export default function ReviewWithComment({ title, description, open, onClose, onSubmit }: ReviewWithCommentProps) {
+export default function ReviewWithComment({
+  title,
+  description,
+  open,
+  onClose,
+  onSubmit,
+  release,
+}: ReviewWithCommentProps) {
+  const { reviews } = useGetReviewRequestsForModel(release.modelId, release.semver, true)
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(reviews[0].model.id)
+
   const [reviewComment, setReviewComment] = useState('')
   const [showError, setShowError] = useState(false)
+  const [selectOpen, setSelectOpen] = useState(false)
+  const [reviewKind, setReviewKind] = useState<ReviewRequestInterface>()
+
+  useEffect(() => {
+    if (reviews) {
+      setReviewKind(reviews[0])
+    }
+  }, [reviews])
 
   function invalidComment() {
     return reviewComment.trim() === '' ? true : false
   }
 
   function submitForm(kind: ResponseTypeKeys) {
     setShowError(false)
-    if (invalidComment() && (kind === ResponseTypes.Reject || kind === ResponseTypes.RequestChanges)) {
+    if (invalidComment() && kind === ResponseTypes.RequestChanges) {
       setShowError(true)
     } else {
-      onSubmit(kind, reviewComment)
+      if (reviewKind) {
+        onSubmit(kind, reviewComment, reviewKind.role)
+      }
     }
   }
 
+  function onChange(_event: React.SyntheticEvent<Element, Event>, newValue: ReviewRequestInterface | null) {
```

Comment:
Super duper picky, but can we use a named import here like we do for everything else?
`import { SyntheticEvent } from 'react'`